ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -335,17 +337,28 @@ private void initializeInstances(List<InstanceConfig> instanceConfigs) throws Ex
       logger.info("Initializing cluster information from {}", dcName);
       for (InstanceConfig instanceConfig : instanceConfigs) {
         int schemaVersion = getSchemaVersion(instanceConfig);
-        if (schemaVersion == 0) {
-          String instanceName = instanceConfig.getInstanceName();
-          logger.info("Adding node {} and its disks and replicas", instanceName);
-          AmbryDataNode datanode =
-              new AmbryDataNode(getDcName(instanceConfig), clusterMapConfig, instanceConfig.getHostName(),
-                  Integer.valueOf(instanceConfig.getPort()), getRackId(instanceConfig), getSslPortStr(instanceConfig));
-          initializeDisksAndReplicasOnNode(datanode, instanceConfig);
-          instanceNameToAmbryDataNode.put(instanceName, datanode);
-          allInstances.add(instanceName);
-        } else {
-          logger.error("Unknown InstanceConfig schema version: {}, ignoring.", schemaVersion);
+        switch (schemaVersion) {
+          case 0:
+            String instanceName = instanceConfig.getInstanceName();
+            long instanceXid = getXid(instanceConfig);
+            if (instanceXid <= currentXid) {
```

Comment:
Are we able to test this method ?  Same for `updateStateOfReplicas ` method.

Reply:
Yes, unit tests are being added. Will update.

Reply:
Added tests.